app/controllers/referral_promos_controller.rb

```diff
@@ -23,8 +23,17 @@ def create
       @publisher.promo_enabled_2018q1 = true
       @publisher.save!
 
-      # TO DO: register channel with promo service
-      # TO DO: create new referral promo from response
+      channels = @publisher.channels.where(verified: true)
+
+      channels.each do |channel|
+        # TO DO: Consider a clever query instead of checking .blank?
+        if channel.referral_promo.blank?
```

Comment:
i'd check also for presence of referral_promo.referral_code

Reply:
consider moving this whole block here (the check if a referral promo needs to be created; and creation of a referral promo) into ChannelPromoRegistrar since all the logic happens together always.

ChannelPromoRegistrar could raise an error if a Channel already has a ReferralPromo or it's is not verified.

Reply:
Thanks, I considered this but wasn't sure where was most appropriate